src/graphics/particle/MagicFlare.cpp

```diff
@@ -359,9 +359,7 @@ void MagicFlareContainer::removeAll() {
 	
 	for(size_t i = 0; i < m_magicFlaresMax; i++) {
 		MagicFlare& flare = m_flares[i];
-		if(flare.exist) {
-			removeFlare(flare);
-		}
+		removeFlare(flare);
```

Comment:
Worried that his might still call io->flarecount-- for already-cleared flares. Perhaps the io member should also be reset in MagicFlare::clear()

Reply:
The flare is made to completely reset in a later commit.